### PR TITLE
feat:Adding prettier as a dev dependency and formatting the whole code

### PR DESCRIPTION
**Pull request by abhishekkumar08**
Fixes:#58

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -26,10 +26,11 @@ Before filling out an issue:
 > We highly recommend that you re-create the bug on [Snack](https://snack.expo.io). If not, list the steps that a reviewer can take to reproduce the behaviour:
 
 Example:
- 
+
 **Snack** : https://snack.expo.io/xxx
 
 - Steps
+
 1. Go to '...'
```
Comment (Uyadav207) [inline .github/ISSUE_TEMPLATE/bug_report.md:34]: Please try making some change, adding spaces doesn't help

File: README.md [hunk 2]
```diff
@@ -32,26 +32,27 @@
...
 
 2. Install NPM packages
+
 ```sh
 npm i
 ```
+
 3.  Run the development server
+
 ```sh
```
Comment (Uyadav207) [inline README.md:52]: only spaces are added to the file?

File: package.json [hunk 3]
```diff
@@ -31,7 +31,8 @@
     "eject": "react-scripts eject",
     "predeploy": "npm run build",
     "deploy": "gh-pages -d build",
-    "postinstall": "patch-package"
+    "postinstall": "patch-package",
+    "format": "prettier --write \"**/*.{js,jsx,json,md}\""
```
Comment (Uyadav207) [inline package.json:35]: please avoid changing the packages.... (can cause merge conflict for others PR's)

File: public/manifest.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
       "src": "logo512.png",
       "type": "image/png",
       "sizes": "512x512"
-    }	    
+    }
```
Comment (Uyadav207) [inline public/manifest.json:19]: * space again ?

File: src/App.js [hunk 5]
```diff
@@ -1,5 +1,8 @@
 import React from "react";
-import { SafeAreaInsetsContext, SafeAreaProvider } from 'react-native-safe-area-context';
+import {
+  SafeAreaInsetsContext,
+  SafeAreaProvider,
+} from "react-native-safe-area-context";
 
 import Root from "./containers";
 import "./App.css";
```
Comment (Uyadav207) [inline src/App.js:8]: This doesn't make any sense. **only spaces have been added** all over the place.

File: src/App.test.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './App';
+import React from "react";
+import ReactDOM from "react-dom";
+import App from "./App";
 
-it('renders without crashing', () => {
-  const div = document.createElement('div');
```
Comment (Uyadav207) [inline src/App.test.js:6]: replacements `" "` to `' '` and vice-a-versa.

File: src/index.js [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
-import 'react-native-safe-area-context';
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import * as serviceWorker from "./serviceWorker";
+import "react-native-safe-area-context";
 
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(<App />, document.getElementById("root"));
```
Comment (Uyadav207) [inline src/index.js:9]: This change will create conflict for other PR's

File: src/serviceWorker.js [hunk 8]
```diff
@@ -93,23 +93,23 @@ function registerValidSW(swUrl, config) {
...
         response.status === 404 ||
-        (contentType != null && contentType.indexOf('javascript') === -1)
+        (contentType != null && contentType.indexOf("javascript") === -1)
       ) {
         // No service worker found. Probably a different app. Reload the page.
-        navigator.serviceWorker.ready.then(registration => {
+        navigator.serviceWorker.ready.then((registration) => {
           registration.unregister().then(() => {
             window.location.reload();
           });
```
Comment (Uyadav207) [inline src/serviceWorker.js:115]: replacements `" "` to `' '` and vice-a-versa.

File: src/serviceWorker.js [hunk 9]
```diff
@@ -121,14 +121,14 @@ function checkValidServiceWorker(swUrl, config) {
...
+        "No internet connection found. App is running in offline mode."
       );
     });
 }
 
 export function unregister() {
-  if ('serviceWorker' in navigator) {
-    navigator.serviceWorker.ready.then(registration => {
+  if ("serviceWorker" in navigator) {
+    navigator.serviceWorker.ready.then((registration) => {
```
Comment (Uyadav207) [inline src/serviceWorker.js:131]: replacements `" "` to `' '` and vice-a-versa.

**Comment by Uyadav207** [comment]
@abhishekkumar08, please have a look at ...... 

- https://reactnativeelements.com/docs/contributing

**Comment by abhishekkumar08** [comment]
> 
> 
> @abhishekkumar08, please have a look at ......
> 
>     * https://reactnativeelements.com/docs/contributing

If you try to understand the PR, it adds prettier to the code base which is a code formatter, I just installed it as a dev dependency and ran the command for it to format the whole code base. Thanks for the review. @Uyadav207

**Comment by Uyadav207** [comment]
@abhishekkumar08, ok if it's prettier also... the changes made are quite huge and this will create `merge conflict`... for the previous PR, as `I have faced the same problem yesterday.` 😢 and `had a hard time resolving conflict`.

- You can check #47 made a `hot-fix` for the conflict.

- Try not to do this.. it causes `syncing the branches and keeping them updated with upstream` difficult. (Talking from experience).

I hope you understand ?

**Comment by abhishekkumar08** [comment]
@Uyadav207 Yeah bro I got your point 😄 . Thanks for the suggestion, any other how should I do this change?

**Review by tewarig — COMMENTED** [review]
but issue #56  #57  and pr made on these are almost the same adding prettier as a dev dependency is not a good idea. we need to run ``` npm run format ``` every time before we make a pr?   
in #56 we are adding a file... which will automatically format the code in vs code if the prettier
 the extension is installed.

**Comment by tewarig** [comment]
> @Uyadav207 Yeah bro I got your point. Thanks for the suggestion, any other way how should I do this change?

just keep an eye on the repo if any pr is merged in this repo make sure you update your repo and format its code again

**Comment by Uyadav207** [comment]
> but issue #56  #57  and pr made on these are almost the same adding prettier as a dev dependency is not a good idea. we need to run ``` npm run format ``` every time before we make a pr?   
> 
> in #56 we are adding a file... which will automatically format the code in vs code if the prettier
> 
>  the extension is installed. 

Exactly.

**Comment by Uyadav207** [comment]
> @Uyadav207 Yeah bro I got your point 😄 . Thanks for the suggestion, any other way how should I do this change?

This is a break change. Need to wait for all PR to get merge than if we push your change.... Then it would not cause any conflict... but syncing the fork then would be necessary for all contributors... Is this making any sense ?

**Comment by jugshaurya** [comment]
@abhishekkumar08 already created an issue and  made a PR about the same thing earlier than you. Kindly work on some other issues.

**Comment by jugshaurya** [comment]
> > @Uyadav207 Yeah bro I got your point smile . Thanks for the suggestion, any other way how should I do this change?
> 
> This is a breaking change. Need to wait for all PR to get merge than if we push your change.... Then it would not cause any conflict... but syncing the fork then would be necessary for all contributors... Is this making any sense ?

@Uyadav207 Initially it may seem a conflicting change but it will be beneficial in long go as developers contributing to the project will be following same styling.

**Comment by tewarig** [comment]
> > > @Uyadav207 Yeah bro I got your point smile . Thanks for the suggestion, any other way how should I do this change?
> > 
> > 
> > This is a breaking change. Need to wait for all PR to get merge than if we push your change.... Then it would not cause any conflict... but syncing the fork then would be necessary for all contributors... Is this making any sense ?
> 
> @Uyadav207 Initially it may seem a conflicting change but it will be beneficial in long go as developers contributing to the project will be following same styling.

yes it will help a lot .

**Pull request closed without merging**